### PR TITLE
Remove sourceMappingURL from JS files

### PR DESCRIPTION
**Pull request by vitvakatu**
This fixes "DevTools failed to load SourceMap" warning in the console

[finishes #180560409]
[ci no changelog needed]

### Pull Request Description

Chrome DevTools automatically try to load [source maps](https://blog.teamtreehouse.com/introduction-source-maps) for each loaded JS file. We do not use source maps at the moment: we neither generate source maps for our JS files, nor bundle existing source maps of our dependencies. Unfortunately, some of our dependencies (notably `firebase`) have a special comment `sourceMappingURL` in their minified code. This special comment fools DevTools that source map is actually provided. This leads to two different (but rather similar) warnings in the console:

In desktop IDE:
Screenshot 2021-12-08 at 18 29 21

In development ("interactive") mode in browser:
image

There are two (or maybe more) possible solutions for the problem:
1. Correctly bundle source maps from our dependencies and potentially generate source maps for our code as well.
2. Remove `sourceMappingURL` comments from our dependencies and ignore source maps.

This PR implements **second** solution. This is the more straightforward and easy way, given that we are not using source maps at the moment and that we do not have any serious amounts of JS code in the codebase. However, it's pretty easy to fall back to the first solution by setting [devtool](https://webpack.js.org/configuration/devtool/) option in our webpack configuration. It's not clear what value for this option we actually want though. (there are like 25+ of them)

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guides.
- [x] All documentation and configuration conforms to the [markdown](https://github.com/enso-org/enso/blob/develop/docs/style-guide/markdown.md) and [YAML](https://github.com/enso-org/enso/blob/develop/docs/style-guide/yaml.md) style guides.
- [x] All code has been tested where possible.

### CODE REVIEW

**Comment by wdanilo** [comment]
Nice research! How does it affect errors from our visualisations? E.g. an user creates a JS visualization and there is an error there – would it be reported correctly in the console (file name, line number)? If not, this should work, as this is a very important user-facing problem.

**Comment by vitvakatu** [comment]
@wdanilo Nice catch, I wasn't even aware that we support fully user-defined visualizations. Luckily, this PR works nicely with them. Here is a screencast where I trigger errors both in built-in (Histogram) visualization, and in user-defined Bubble visualization. You can't see the exact file location in the console, but DevTools shows you the entire source code of visualization after clicking.


https://user-images.githubusercontent.com/6566674/145563805-4139277f-75a4-4a28-9005-ef659b1013b9.mov

**Comment by wdanilo** [comment]
@vitvakatu that's awesome! Would you be so nice and explain to me WHY does it work? We need source maps to do it, so how on earth does it still work correctly?

**Review by wdanilo — CHANGES_REQUESTED** [review]
@vitvakatu I checked again your PR and unfortunately it does not work. Take a look at the screenshot from the video that you've posted:

image

Because of the lack of source maps, the location does not contain the file name nor the line number. I know that after clicking it you've been able to see the code source, but:
1. The source line numbers were not the same as in the original JS file.
2. This will not work as soon as we enable JS code minification (TBH, I'm surprised it is not enabled by default).

To sum this up, source mapping is NOT working for user-defined visualisations and this is bad and needs to be fixed. I'm totally OK with removing source map info from some 3rd party libs, but this needs to work for our viz system.

**Comment by vitvakatu** [comment]
@wdanilo

**My opinion changed since I wrote this post, please see https://github.com/enso-org/enso/pull/3187#issuecomment-993429916**

#### Why this PR should be merged

TL;DR: this PR fixes a really narrow issue with `warnings` in the console. It does not change how our visualizations work, nor enables source maps generation for our files. Our visualizations have a problem with showing their source code, but it is caused by the way our visualizations are implemented at the moment. I suggest merging this PR, and solving visualization problems in a separate task (which will require some research).

> Would you be so nice and explain to me WHY does it work? We need source maps to do it, so how on earth does it still work correctly?

DevTools are clever enough to be able to show the source code of every dynamically-created [`Function`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/Function) object. This is the way we create visualizations, by loading and executing some JS code (it's similar to using `eval`). So yeah, source maps are not needed in this case.

> I checked again your PR and unfortunately, it does not work.

This is not true. This PR does not introduce any new regressions, as far as I'm concerned, and doesn't change the situation around exceptions in visualizations at all. It is true that we have a problem with that, but it should be solved in a separate task I believe (see below).

> Because of the lack of source maps, the location does not contain the file name nor the line number.

Yep, I mentioned it in my comment with the screencast, that's unfortunate. But it is not caused by this PR, this has been always working this way.

> This will not work as soon as we enable JS code modification

I think there is no correlation between code modification done by Webpack and our visualizations, because they are not included in the Webpack pipeline. I doubt we would ever use code modification for our visualization in the future. Again, see below.

#### On problem with visualizations

Our visualizations are implemented by loading (in compile time for built-in visualizations, and in run time for user-defined visualizations) some JS code and creating a JS `Function` object. The source code of visualizations does not participate in the packaging process (by Webpack): it's not needed for built-in visualizations, and it's actually impossible for user-defined ones because they are located in Enso's project directory. 

This means, that generating and providing the source maps is not possible with the current design. To change that, we need to set up the source map generation process (it would be harder for user-defined visualizations, because we do not have Webpack there) and preprocess all JS files, including `sourceMappingURL` (inline version of source map) into them. I'm not sure that DevTools will understand to look at the inline source map in this case though, it needs further investigation.

See [a created task](https://www.pivotaltracker.com/story/show/180620124).

**Comment by vitvakatu** [comment]
Since our demo day, I was thinking if it actually makes sense to go with the first solution: load source maps from our dependencies correctly, and also generate them for our own sources. In this case, this PR should probably be reworked. The reason for that is that source maps can be useful when debugging performance issues with our new profiling tools, as @MichaelMauderer pointed. 

I'm converting this PR to draft until we come to some conclusion here (and I'll rework it if it's needed).

**Comment by wdanilo** [comment]
@vitvakatu one question – I understand that your solution does NOT change in any way how users would be able to define / reload visualisations? We do not want to use web pack or any pre-processor for user-defined visualisations. We can generate source maps for them with our Rust scripts, but this is super mportant for users to be able to change visualisations code and reload them without restarting Enso as fast as possible (waiting a few seconds for web pack is a no-go).

**Comment by vitvakatu** [comment]
@wdanilo yeah, I was considering Webpack initially, but it doesn't work nicely with user-defined visualizations, and it doesn't work with built-in visualizations either (because of complicated multistage build). So I plan to generate source maps in Rust, hopefully, existing libraries will help me with that.

**Comment by vitvakatu** [comment]
Will reopen after rework
